lib/ClientConnectContext.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Amp\Socket;
+
+use function Amp\Socket\Internal\normalizeBindToOption;
+
+final class ClientConnectContext {
+    private $bindTo = null;
+    private $connectTimeout = 10000;
+    private $maxAttempts = 2;
```

Comment:
Yes, that's exactly the thing this setting tries to limit, otherwise you can make connection attempts hang for a long time.